src/pricer/pricer.ts

```diff
@@ -0,0 +1,86 @@
+import { BigNumber as BigNumberJs } from 'bignumber.js';
+import log from 'loglevel';
+import {
+  ExchangeApiName,
+  INTERMEDIATE_CURRENCY,
+  tokenToApi,
+  createExchangeApi,
+} from './utils';
+
+const getExchangeRate = async (
+  sourceCurrency: string,
+  targetCurrency: string,
+  intermediateCurrency = INTERMEDIATE_CURRENCY
+) => {
+  log.debug('RelayPricer: intermediary Currency', intermediateCurrency);
+
+  const sourceExchangeRate = await queryExchangeRate(
+    sourceCurrency,
+    intermediateCurrency
+  );
+
+  const targetExchangeRate = await queryExchangeRate(
+    targetCurrency,
+    intermediateCurrency
+  );
+
+  if (sourceExchangeRate.isEqualTo(0) || targetExchangeRate.isEqualTo(0)) {
+    const message = `Currency conversion for pair ${sourceCurrency}:${targetCurrency} not found in current exchange api`;
+    log.error(`RelayPricer exchange rates: ${message}`);
+    throw Error(message);
+  }
+
+  log.debug(
+    'RelayPricer exchange rates:',
+    sourceExchangeRate.toString(),
+    targetExchangeRate.toString()
+  );
+
+  return sourceExchangeRate.dividedBy(targetExchangeRate);
+};
+
+const queryExchangeRate = async (
+  sourceCurrency: string,
+  targetCurrency: string
+): Promise<BigNumberJs> => {
+  const exchangeApis = tokenToApi[sourceCurrency.toUpperCase()];
+
+  if (!exchangeApis) {
+    throw Error(`Token ${sourceCurrency} does not have an API to query`);
+  }
+
+  log.debug(`${exchangeApis.toString()} found for token ${sourceCurrency}`);
+
+  return queryExchangeApis(exchangeApis, sourceCurrency, targetCurrency);
+};
+
+const queryExchangeApis = async (
+  exchangesApi: ExchangeApiName[],
+  sourceCurrency: string,
+  targetCurrency: string
+): Promise<BigNumberJs> => {
+  for (const api of exchangesApi) {
+    const exchangeApi = createExchangeApi(api);
+
+    try {
+      const exchangeRate = await exchangeApi.queryExchangeRate(
+        sourceCurrency,
+        targetCurrency
+      );
+
+      if (exchangeRate) {
+        tokenToApi[sourceCurrency.toUpperCase()] = [
+          ...new Set([api, ...exchangesApi]),
+        ];
+
+        return exchangeRate;
+      }
+    } catch (e: unknown) {
+      log.error(e);
+    }
+  }
+
+  return BigNumberJs(0);
+};
```

Comment:
I'd simplify this a bit by removing the try/catch
```suggestion
  for (const api of exchangesApi) {
    const exchangeApi = createExchangeApi(api);


      const exchangeRate = await exchangeApi.queryExchangeRate(
        sourceCurrency,
        targetCurrency
      ).catch((e: unknown) => {
        log.error(e);
        
        return undefined;
      });

      if (exchangeRate) {
        tokenToApi[sourceCurrency.toUpperCase()] = [
          ...new Set([api, ...exchangesApi]),
        ];

        return exchangeRate;
      }
      
      log.info(`No exchange rate found for ${sourceCurrency}/${targetCurrency} using API ${api}`); // or log.debug, I'm not sure 😅
  }

  return BigNumberJs(0);
};
```

Reply:
I think that here we could include a notifier as well for the grafana logs, so we could know if an API failed and validate if something needs to be changed to make it work again.

Reply:
@franciscotobar I think this is a good idea, let's change the format to be similar to the [one used in the server](https://github.com/rsksmart/rif-relay-server/blob/320b81165f21020b81b712e817833f69dc48743b/src/ReplenishFunction.ts#L9-L10).
After doing it, we should get @david-iov involved to create the related notification.